drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -53,6 +54,27 @@ local function device_removed(driver, device)
   log.info("device removed")
 end
 
+-- New profiles need to be added for devices that have more switch endpoints
+local MAX_MULTI_SWITCH_EPS = 7
+local function do_configure(driver, device)
+  -- Note: This profile switching is needed because of shortcoming in the generic fingerprints
+  -- where devices with multiple endpoints with the same device type cannot be detected
+  local switch_eps = device:get_endpoints(clusters.OnOff.ID)
+  local num_switch_eps = #switch_eps
+  table.sort(switch_eps)
+  --Default MCD switch handling depends on consecutive endpoint numbering
+  if num_switch_eps == switch_eps[num_switch_eps] then
+    if num_switch_eps > 1 and num_switch_eps <= MAX_MULTI_SWITCH_EPS then
+      device:try_update_metadata({profile = string.format("switch-%d", num_switch_eps)})
+    elseif num_switch_eps > MAX_MULTI_SWITCH_EPS then
+      error(string.format(
+        "Matter multi switch device will not function. Profile doesn't exist with %d components",
+        num_switch_eps
```

Comment:
In theory we could still support the first 7. I'm guessing you thought about this and decided not to, but I was wondering your reasoning.

Reply:
My main reason was to emit an error that would be easy to spot to know if we should add more profiles. But considering what you're suggesting, we could emit the error and update the profile to something that would work more. I will double check what would happen in a case where we support the first 7 and start getting reports from ep num 8. I want to make sure it doesn't cause more problems than it solves

Reply:
I did what you suggested. What will happen is that reports from the endpoints higher than the max we support will be mapped to the main component. The error will be in the logs and hub_logs, so that will at least give us a clue tht it is happening

Reply:
your error message will never fire, as is

`num_switch_eps > 1` is still true for `num_switch_eps > 8`, so the second branch is never entered